smart_contracts/xgov_registry/contract.py

```diff
@@ -361,6 +365,59 @@ def create(self) -> None:
         self.xgov_manager.value = arc4.Address(Txn.sender)
         assert self.entropy() == TemplateVar[Bytes]("entropy")
 
+    @arc4.abimethod()
+    def init_proposal_contract(self, size: arc4.UInt64) -> None:
+        """
+        Initializes the Proposal Approval Program contract.
+
+        Args:
+            size (arc4.UInt64): The size of the Proposal Approval Program contract
+
+        Raises:
+            err.UNAUTHORIZED: If the sender is not the current xGov Manager
+        """
+
+        assert self.is_xgov_manager(), err.UNAUTHORIZED
+
+        box, exist = self.proposal_approval_program.maybe()
+        if exist:
+            self.proposal_approval_program.resize(size.native)
+        else:
+            # Initialize the Proposal Approval Program contract
+            self.proposal_approval_program.create(size=size.native)
+
+    @arc4.abimethod()
+    def load_proposal_contract(self, offset: arc4.UInt64, data: Bytes) -> None:
+        """
+        Loads the Proposal Approval Program contract.
+
+        Args:
+            offset (arc4.UInt64): The offset in the Proposal Approval Program contract
+            data (Bytes): The data to load into the Proposal Approval Program contract
+
+        Raises:
+            err.UNAUTHORIZED: If the sender is not the current xGov Manager
+        """
+
+        assert self.is_xgov_manager(), err.UNAUTHORIZED
+
+        # Load the Proposal Approval Program contract
+        self.proposal_approval_program.replace(start_index=offset.native, value=data)
+
+    @arc4.abimethod()
+    def delete_proposal_contract_box(self) -> None:
+        """
+        Deletes the Proposal Approval Program contract box.
+
+        Raises:
+            err.UNAUTHORIZED: If the sender is not the current xGov Manager
+        """
+
+        assert self.is_xgov_manager(), err.UNAUTHORIZED
+
+        # Delete the Proposal Approval Program contract box
+        self.proposal_approval_program.delete()
```

Comment:
Just a sanity check: can we not have one `load_proposal_contract` method that creates the box if it's the first time or resizes/overwrites it if it's an update?

Reply:
Agreed, IRL, this might be future optimization.